pulsar-client/src/main/java/org/apache/pulsar/client/impl/ZeroQueueConsumerImpl.java

```diff
@@ -84,11 +84,20 @@ protected Message<T> internalReceive() throws PulsarClientException {
 
     @Override
     protected CompletableFuture<Message<T>> internalReceiveAsync() {
+        // Just being cautious
+        if (incomingMessages.size() > 0) {
+            log.error("The incoming message queue should never be greater than 0 when Queue size is 0");
+            incomingMessages.forEach(Message::release);
+            incomingMessages.clear();
+        }
```

Comment:
> Zero queue consumer never caches messages in `incomingMessages`

This could happen in `internalReceive()`. 
So when `internalReceiveAsync()` and `internalReceive()` are called concurrently, we should not do this check.

Reply:
How could it happen? `internalReceive()` only calls `fetchSingleMessageFromBroker`, which is overridden in `ZeroQueueConsumerImpl` and won't add any `Message` object into `incomingMessages`.

Reply:
@BewareMyPower This?
https://github.com/apache/pulsar/blob/1a098d5418be7d4674f9ba9dbc4da0018f91654a/pulsar-client/src/main/java/org/apache/pulsar/client/impl/ZeroQueueConsumerImpl.java#L114

Reply:
@Jason918  Because `internalReceive ` already checks this behaviour, I think we can remove this check directly.

Reply:
@BewareMyPower  I changed the test, PTAL :)